aten/src/ATen/BatchingRegistrations.cpp

```diff
@@ -694,6 +695,53 @@ Tensor new_empty_batching_rule(
   return physical_view.newLogicalFromPhysical(result);
 }
 
+Tensor new_empty_strided_batching_rule(
+    const Tensor& self,
+    IntArrayRef size,
+    IntArrayRef stride,
+    const TensorOptions& options) {
+  auto physical_view = MultiBatchVmapTransform::logicalToPhysical(self);
+  auto physical_size = physical_view.getPhysicalShape(size);
+
+  // Let [B0, B1, B2] be the shape of the batch dims. We know what the physical
```

Comment:
What happens if all the batch dimensions are not at the front?

Reply:
MultiBatchVmapTransform::logicalToPhysical(self) moved all of the batch dims to the front

Reply:
But I should clarify in the comment that we are going to create the batch dimensions at the front, even if in the original tensors being vmapped the batch dim wasn't at the front.